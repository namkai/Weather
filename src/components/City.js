import React from 'react';

const City = ({ name }) => {
  return (
	  <li className="nav-item">
		  <a className="nav-link" href="#">{name}</a>
	  </li>
  );
};

export default City;