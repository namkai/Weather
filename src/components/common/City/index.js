import Chart from './view';

export default Chart;