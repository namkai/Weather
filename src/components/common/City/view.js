import PropTypes from 'prop-types';
import React from 'react';

const City = ({ name, fetchWeather }) => (
  <li className="nav-item">
    <a className="nav-link" onClick={() => fetchWeather(name)} href="#">{name}</a>
  </li>
);

City.propTypes = {
  name: PropTypes.string,
  fetchWeather: PropTypes.func,
};

export default City;
