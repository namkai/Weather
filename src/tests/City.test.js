import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import City from '../components/common/City';

it('renders correctly', () => {
	const city = shallow(<City />);
	expect(toJson(city)).toMatchSnapshot();
});