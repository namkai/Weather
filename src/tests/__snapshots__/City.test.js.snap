// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link"
    href="#"
    onClick={[Function]}
  />
</li>
`;
